spanner/spannertest/db.go

```diff
@@ -1090,3 +1147,40 @@ func parseAsDate(s string) (civil.Date, error) { return civil.ParseDate(s) }
 func parseAsTimestamp(s string) (time.Time, error) {
 	return time.Parse("2006-01-02T15:04:05.999999999Z", s)
 }
+
+// getIDsForGeneratedExpression returns a list of column names the expression depends on
+func getIDsForGeneratedExpression(e spansql.Expr) []spansql.ID {
```

Comment:
I'm not sure this is the best way to do this from a maintenance and completeness perspective. Instead I think it might be better to edit the `evalContext.evalExpr` function to also return the id's that the expression needs to be evaluated. That would keep the logic for what is needed for a certain expression in one place, and reduce the chance of missing something, and also ensure that whenever we add support for new expressions, those are also automatically included in this array.
This implementation currently does not detect any dependencies on other columns if they are used in:
1. Function calls
2. Array expressions

Reply:
Yeah I really didn't like this approach, it felt like it was prone to error and future failure but couldn't figure out something smarter. I think your suggestion sounds solid I can check into that, So adding a 3rd return argument to evalExpr, potentially returning and error and the list of dependencies?

Reply:
Yes, something like that was my thought as well.